### PR TITLE
Added Emacs support

### PR DESCRIPTION
**Pull request by zefirka**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ Supported editors:
 - [PhpStorm](https://www.jetbrains.com/phpstorm/)
 - [IDEA 14 Community Edition](https://www.jetbrains.com/idea/download/)
 - [Vim](http://www.vim.org/)
+- [Emacs](https://www.gnu.org/software/emacs/)
```
Comment (lahmatiy) [inline README.md]: It also should be added to `editor` setting value list (https://github.com/lahmatiy/open-in-editor#editor)

File: lib/utils.js [hunk 2]
```diff
@@ -1,6 +1,14 @@
 var osHomeDir = require('os-homedir')();
 var path = require('path');
 
+var osascript = function(script) {
```
Comment (lahmatiy) [inline lib/utils.js]: Let's move this stuff to `common/terminal.js` or something like that

File: README.md (hunk 1 above)
Comment (zefirka) [inline README.md]: Done

File: lib/utils.js (hunk 2 above)
Comment (zefirka) [reply lib/utils.js]: Ok. Moved everything regarding terminal interaction from `utils.js`  to `common/terminal.js`.

File: lib/utils.js [hunk 3]
```diff
@@ -49,5 +49,15 @@ module.exports = {
   },
   append: function(str, appendix) {
     return String(str).replace(/\s*$/, (str ? ' ' : '') + appendix);
+  },
+  detectFor: function (platforms, editor) {
+    return function() {
+      // we use only system parts so we haven't to check anything except os
+      if (~platforms.indexOf(process.platform)) {
```
Comment (lahmatiy) [inline lib/utils.js]: `~platforms.indexOf(process.platform)` -> `platforms.indexOf(process.platform) !== -1`, please

File: lib/utils.js [hunk 4]
```diff
@@ -49,5 +49,15 @@ module.exports = {
   },
   append: function(str, appendix) {
     return String(str).replace(/\s*$/, (str ? ' ' : '') + appendix);
+  },
+  detectFor: function (platforms, editor) {
```
Comment (lahmatiy) [inline lib/utils.js]: let's rename it to `checkPlatformSupport`

File: lib/common/terminal.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+var terminal = function(cmd) {
+  return 'tell application "Terminal" to do script "' + cmd + '"';
+};
+
+var runInTerminal = function (cmd) {
+  return osascript(terminal('cd {projectPath}; ' + cmd));
+};
+
+module.exports = {
+  osascript: osascript,
```
Comment (lahmatiy) [inline lib/common/terminal.js]: Why do we need to expose `osascript` and `terminal`? I think `module.exports = runInTerminal;` is enough.

**Comment by lahmatiy** [comment]
@zefirka Hi! Can you fix last comments?

**Comment by zefirka** [comment]
@lahmatiy Sure! It's done.

**Comment by lahmatiy** [comment]
Thank you!

**Pull request merged**